runtime/cudaq/algorithms/state.cpp

```diff
@@ -39,7 +39,12 @@ std::complex<double> state::operator[](std::size_t idx) {
   if (shape.size() != 1)
     throw std::runtime_error("Cannot request 1-d index into density matrix. "
                              "Must be a state vector.");
-  return stateData[idx];
+  std::size_t numQubits = std::log2(stateData.size());
+  unsigned int newIdx = 0;
+  for (auto i = 0u; i < numQubits; ++i)
+    if (idx & (1 << i))
+      newIdx |= (1u << ((numQubits - 1) - i));
```

Comment:
I believe `newIdx` and `i` should be declared as `std::size_t` in order to support >32 qubits.